Kotlin/lib/src/main/java/com/wolt/blurhashkt/BlurHashDecoder.kt

```diff
@@ -94,30 +91,30 @@ object BlurHashDecoder {
         return arrayOf(sRgbToLinear(intR), sRgbToLinear(intG), sRgbToLinear(intB))
     }
 
-    fun sRgbToLinear(value: Int): Float {
+    private fun sRgbToLinear(value: Int): Float {
         val v = value / 255f
-        return if (v <= 0.04045) (v / 12.92f) else (pow((v + 0.055) / 1.055, 2.4).toFloat())
+        return if (v <= 0.04045) (v / 12.92f) else ((v + 0.055) / 1.055).pow(2.4).toFloat()
     }
 
     private fun decodeAc(value: Int, maximumValue: Float): Array<Float> {
         val quantR = value / (19 * 19)
         val quantG = (value / 19) % 19
         val quantB = value % 19
         return arrayOf(
-                signPow((quantR - 9) / 9.0, 2.0) * maximumValue,
-                signPow((quantG - 9) / 9.0, 2.0) * maximumValue,
-                signPow((quantB - 9) / 9.0, 2.0) * maximumValue
+                signPow((quantR - 9) / 9.0f, 2.0f) * maximumValue,
+                signPow((quantG - 9) / 9.0f, 2.0f) * maximumValue,
+                signPow((quantB - 9) / 9.0f, 2.0f) * maximumValue
         )
     }
 
-    private fun signPow(value: Double, exp: Double) = copySign(pow(abs(value), exp), value).toFloat()
+    private fun signPow(value: Float, exp: Float) = value.pow(exp).withSign(value)
```

Comment:
`abs()` is not needed. Yes, the sign may be different after the `pow`, but `withSign` (Kotlin for `copySign`) changes it again anyway.